bee-common/bee-common/src/packable/unpacker.rs

```diff
@@ -0,0 +1,20 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use super::UnpackError;
+
+/// A type that can unpack any value that implements `Packer`.
+pub trait Unpacker {
+    /// The error type that can be returned if some error occurs while unpacking.
+    type Error: UnpackError;
+
+    /// Unpack a sequence of bytes from the `Unpacker`.
+    fn unpack_bytes(&mut self, slice: &mut [u8]) -> Result<(), Self::Error>;
+
+    /// Unpack a statically-sized sequence of bytes from the `Unpacker`.
+    fn unpack_exact_bytes<const N: usize>(&mut self) -> Result<[u8; N], Self::Error> {
```

Comment:
This function is presumably only used for numeric types, and those are implemented with a macro. Why not put this impl in the macro? I can't think of anything else for which it would be useful.